google-cloud-pubsub/lib/google/cloud/pubsub/errors.rb

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/errors"
+
+module Google
+  module Cloud
+    module PubSub
+      ##
+      # Indicates that the {AsyncPublisher} has been stopped and cannot accept
+      # messages to publish.
+      #
+      class AsyncPublisherStopped < Google::Cloud::Error
```

Comment:
Would it be better to create and inherit these classes from `Google::Cloud::PubSub::Error`?